charm-nextcloud/src/charm.py

```diff
@@ -166,7 +172,15 @@ def _on_cluster_relation_changed(self, event):
     
     def _on_cluster_relation_departed(self, event):
         logger.debug("!!!!!!!!cluster relation departed!!!!!!!!")
-        logger.debug("Unit {} left the cluster :(".format(event.unit.name))
+        if self.model.unit.is_leader():
+            logger.debug("Unit {} left the cluster :(".format(event.unit.name))
+            if not self._stored.nextcloud_initialized:
+                event.defer()
+                return
+            self.framework.breakpoint('departed')
+            # TODO: Seems like the unit data is gone already. Which mean I can't get hold if its ingress-address to remove it from trusted domains...
+            #peer_ingress_address = event.relation.data.get(event.unit).get('ingress-address')
```

Comment:
seems like the event here does not contain the relatation data anymore.. How can I get the IP from the unit before it leaves?? Do I need to set it on the relation on the worker side first, for the leader so get it?

Reply:
Like this:

if leader:
  read worker IP or defer()
if not leader:
  set IP in relation data